Riot/Modules/Room/VoiceMessages/VoiceMessagePlaybackView.swift

```diff
@@ -0,0 +1,110 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+protocol VoiceMessagePlaybackViewDelegate: AnyObject {
+    func voiceMessagePlaybackViewDidRequestToggle()
+}
+
+struct VoiceMessagePlaybackViewDetails {
+    var currentTime: String = ""
+    var progress = 0.0
+    var samples: [Float] = []
+    var playing: Bool = false
+    var playbackEnabled = false
+    var recording: Bool = false
+}
+
+class VoiceMessagePlaybackView: UIView {
+    
+    private var waveformView: VoiceMessageWaveformView!
+    
+    @IBOutlet private var backgroundView: UIView!
+    @IBOutlet private var recordingIcon: UIView!
+    @IBOutlet private var playButton: UIButton!
+    @IBOutlet private var elapsedTimeLabel: UILabel!
+    @IBOutlet private var waveformContainerView: UIView!
+    
+    weak var delegate: VoiceMessagePlaybackViewDelegate?
+    
+    var details: VoiceMessagePlaybackViewDetails?
+    
+    static func instanceFromNib() -> VoiceMessagePlaybackView {
+        let nib = UINib(nibName: "VoiceMessagePlaybackView", bundle: nil)
+        guard let view = nib.instantiate(withOwner: nil, options: nil).first as? Self else {
+          fatalError("The nib \(nib) expected its root view to be of type \(self)")
+        }
+        return view
+    }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(handleThemeDidChange), name: .themeServiceDidChangeTheme, object: nil)
+        
+        backgroundView.layer.cornerRadius = 12.0
+        
+        waveformView = VoiceMessageWaveformView(frame: waveformContainerView.bounds)
+        waveformContainerView.vc_addSubViewMatchingParent(waveformView)
+    }
+    
+    func configureWithDetails(_ details: VoiceMessagePlaybackViewDetails?) {
+        guard let details = details else {
+            return
+        }
+        
+        playButton.isEnabled = details.playbackEnabled
+        playButton.isHidden = details.recording
+        recordingIcon.isHidden = !details.recording
+        elapsedTimeLabel.text = details.currentTime
+        waveformView.progress = details.progress
+        
+        if ThemeService.shared().isCurrentThemeDark() {
+            playButton.setImage((details.playing ? Asset.Images.voiceMessagePauseButtonDark.image : Asset.Images.voiceMessagePlayButtonDark.image), for: .normal)
+            backgroundView.backgroundColor = UIColor(rgb: 0x394049)
```

Comment:
We should not use hardcoded colors but use colors from theme, i.e. here it's `theme.colors.tile`.